File: ru.kmorozov.gbd.core/src/main/java/ru/kmorozov/gbd/core/logic/Proxy/EmptyProxyListProvider.java
package ru.kmorozov.gbd.core.logic.Proxy;

public class EmptyProxyListProvider extends AbstractProxyListProvider {

    public static final EmptyProxyListProvider INSTANCE = new EmptyProxyListProvider();

    private EmptyProxyListProvider() {
        proxyList.add(HttpHostExt.NO_PROXY);

        proxyListCompleted.set(true);
    }

    @Override
    public void updateProxyList() {

    }
}

File: ru.kmorozov.gbd.core/src/main/java/ru/kmorozov/gbd/core/logic/Proxy/EmptyProxyListProvider.kt
package ru.kmorozov.gbd.core.logic.Proxy

class EmptyProxyListProvider private constructor() : AbstractProxyListProvider() {

    init {
        proxyList.add(HttpHostExt.NO_PROXY)

        proxyListCompleted.set(true)
    }

    override fun updateProxyList() {

    }

    companion object {

        val INSTANCE = EmptyProxyListProvider()
    }
}
